vulnerabilities/importers/rockylinux.py

```diff
@@ -0,0 +1,199 @@
+        try:
+            response = requests_session.get(current_url)
+            if response.status_code != requests.codes.ok:
+                logger.error(f"Failed to fetch RedHat CVE results from {current_url}")
+                break
+            cve_data = response.json().get("advisories") or []
+            cve_data_list.extend(cve_data)
+        except Exception as e:
+            logger.error(f"Failed to fetch rockylinux CVE results from {current_url} {e}")
+            break
+        if not cve_data:
+            break
+        page_no += 1
+    return cve_data_list
+
+
+class RockyLinuxImporter(Importer):
+    spdx_license_expression = "CC-BY-4.0"
+    license_url = "https://access.redhat.com/security/data"
+    importer_name = "Rocky Importer"
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+
+        for rockylinux_cve in fetch_cves():
+            yield to_advisory(rockylinux_cve)
+
+
+def to_advisory(advisory_data):
+    aliases = advisory_data.get("name") or ""
+    date_published = dateparser.parse(advisory_data.get("publishedAt", ""))
+
+    summary = advisory_data.get("description") or ""
+    affected_products = advisory_data.get("affectedProducts") or []
+    affected_packages = []
+    for products in affected_products:
+        packages = advisory_data["rpms"][products]["nvras"]
+        affected_packages.extend(packages)
+    processed_affected_packages: List[AffectedPackage] = []
+    for rpm in affected_packages:
+        purl = rpm_to_purl(rpm_string=rpm.rsplit(".rpm", 1)[0] or "", namespace="rocky-linux")
+        if purl:
+            try:
+                affected_version_range = RpmVersionRange.from_versions(sequence=[purl.version])
+                processed_affected_packages.append(
+                    AffectedPackage(
+                        package=PackageURL(
+                            type=purl.type,
+                            name=purl.name,
+                            namespace=purl.namespace,
+                            qualifiers=purl.qualifiers,
+                            subpath=purl.subpath,
+                        ),
+                        affected_version_range=affected_version_range,
+                        fixed_version=None,
+                    )
+                )
+            except Exception as e:
+                logger.error(f"Failed to parse version range {purl.version} for {purl} {e}")
+
+    references = [
+        Reference(
+            severities=[], url=fix.get("sourceLink") or "", reference_id=fix.get("ticket") or ""
+        )
+        for fix in advisory_data["fixes"]
+    ]
+
+    for ref in advisory_data.get("cves") or []:
+
+        name = ref.get("name", "")
+        if not isinstance(name, str):
+            logger.error(f"Invalid advisory type {name}")
+            continue
+
+        if "CVE" in name.upper():
+            severity_vector_pattern = r"CVSS:3\.1/([A-Z:/]+)"
```

Comment:
@ambuj-1211 Why are you using pattern matching? If your goal is to avoid `UNKNOWN`, you should use an `if` statement instead, as pattern matching can be time-consuming.

Reply:
using pattern matching because the cwe strings were also like this (CWE-273, CWE-123, CWE-546) or they may have other things too like CWE-123 and CWE-345

Reply:
because of this reason I used pattern matching I think if condition won't work what do you say?